### PR TITLE
ASoC: SOF: prefix some terse and cryptic dev_dbg() with __func__

### PR DESCRIPTION
**Pull request by marc-hb**
These helped troubleshoot some DMA issue in SOF.

Signed-off-by: Marc Herbert <marc.herbert@intel.com>

### CODE REVIEW

File: sound/hda/trace.h [hunk 1]
```diff
@@ -65,7 +65,7 @@ DECLARE_EVENT_CLASS(hdac_stream,
 		__entry->stream_tag = (azx_dev)->stream_tag;
 	),
 
-	TP_printk("stream_tag: %d", __entry->stream_tag)
+	TP_printk("%s: stream_tag: %d", __func__, __entry->stream_tag)
```
Comment (ranj063) [inline sound/hda/trace.h]: I'm not sure about this one. Doesn't a tracepoint already record the function name?

Comment (marc-hb) [reply sound/hda/trace.h]: Thanks @ranj063 , removed.

File: sound/soc/sof/trace.c [hunk 2]
```diff
@@ -480,7 +480,8 @@ int snd_sof_init_trace(struct snd_sof_dev *sdev)
 		goto table_err;
 
 	sdev->dma_trace_pages = ret;
-	dev_dbg(sdev->dev, "dma_trace_pages: %d\n", sdev->dma_trace_pages);
+	dev_dbg(sdev->dev, "%s: dma_trace_pages: %d\n",
+		__func__, sdev->dma_trace_pages);
```
Comment (ranj063) [inline sound/soc/sof/trace.c:484]: I checked the SOF directory and all of the above are unique debug messages. Do you really think we need the function name to locate them?

Comment (marc-hb) [reply sound/soc/sof/trace.c:484]: I found the messages mostly useless and the function names very useful but I guess it depends on your expertise and what you're looking for. Having to (repeatedly?) `git grep` every log line is not very productive.

The one on line 420 is especially meaningless.

**Review by marc-hb — COMMENTED** [review]
For the record I did this while debugging SOF issue https://github.com/thesofproject/sof/issues/4356

@ranj063 what are the specific drawbacks of adding `__func__` that concern you?

I just checked and `__func__` is currently used in 31000 places in the kernel.

**Comment by ranj063** [comment]
> For the record I did this while debugging SOF issue [thesofproject/sof#4356](https://github.com/thesofproject/sof/issues/4356)
> 
> @ranj063 what are the specific drawbacks of adding `__func__` that concern you?
> 
> I just checked and `__func__` is currently used in 31000 places in the kernel.

I dont have a problem with __func__but my concern is that this will be hard to upstream if it doesnt bring in additional value than what the existing debug messages already provide.

File: sound/soc/sof/trace.c (hunk 2 above)
Comment (plbossart) [reply sound/soc/sof/trace.c:484]: I don't disagree with @marc-hb, doing a grep on e.g. period_bytes isn't great really.

**Review by plbossart — APPROVED** [review]

**Comment by plbossart** [comment]
> > For the record I did this while debugging SOF issue [thesofproject/sof#4356](https://github.com/thesofproject/sof/issues/4356)
> > @ranj063 what are the specific drawbacks of adding `__func__` that concern you?
> > I just checked and `__func__` is currently used in 31000 places in the kernel.
> 
> I dont have a problem with __func__but my concern is that this will be hard to upstream if it doesnt bring in additional value than what the existing debug messages already provide.

it's not so bad here, there's actually on-going work to make the results of the dmesg logs scriptable, it's much better to add some locality information that includes device+function for uniqueness.

**Review by ranj063 — APPROVED** [review]

**Comment by marc-hb** [comment]
> it's not so bad here, there's actually on-going work to make the results of the dmesg logs scriptable,

Do you mean this? https://lwn.net/Articles/857148/ printk indexing

**Comment by plbossart** [comment]
> > it's not so bad here, there's actually on-going work to make the results of the dmesg logs scriptable,
> 
> Do you mean this? https://lwn.net/Articles/857148/ printk indexing

yep. And FWIW in the SOF trace we implement this notion of indexing all the messages so that it can be compressed. There is definitely some merit in having unique messages in different locations.

**Comment by marc-hb** [comment]
I don't know how it works in the new kernel thing. For sof-logger the only index is a file offset so the messages don't need to be unique: https://github.com/thesofproject/sof/blob/5374da8c8917e5/tools/logger/convert.c#L597

**Comment by plbossart** [comment]
SOFCI TEST

**Pull request merged**